src/core/file.cc

```diff
@@ -1078,8 +1078,8 @@ make_file_impl(int fd, file_open_options options, int flags, struct stat st) noe
     auto st_dev = st.st_dev;
     static thread_local std::unordered_map<decltype(st_dev), internal::fs_info> s_fstype;
 
-    future<> get_fs_info = s_fstype.count(st_dev) ? make_ready_future<>() :
-        engine().fstatfs(fd).then([fd, st_dev] (struct statfs sfs) {
+    if (s_fstype.count(st_dev) == 0) {
```

Comment:
could also use 
```c++
if (!s_fstype.contains(st_dev)) {
```

since we dropped the support of C++17.

Reply:
Yeah, this patch keeps finding method as is, next patch fixes it